src/libcore/num/dec2flt/mod.rs

```diff
@@ -122,11 +122,35 @@ macro_rules! from_str_float_impl {
             /// * '2.5E10', or equivalently, '2.5e10'
             /// * '2.5E-10'
             /// * '5.'
-            /// * '.5', or, equivalently,  '0.5'
+            /// * '.5', or, equivalently, '0.5'
             /// * 'inf', '-inf', 'NaN'
             ///
             /// Leading and trailing whitespace represent an error.
             ///
+            /// # Grammar
+            ///
+            /// All strings that adhere to the following regular expression
+            /// will result in an [`Ok`] being returned:
+            ///
+            /// ```
+            /// (\+|-)?
+            /// (inf|
+            ///  NaN|
+            ///  ([0-9]+|
+            ///   [0-9]+\.[0-9]*|
+            ///   [0-9]*\.[0-9]+)
+            ///  ((e|E)
+            ///   (\+|-)?
+            ///   [0-9]+)?)
+            /// ```
```

Comment:
I'm not tied to the particular formatting of the regular expression here. You can visualize this [as a railroad diagram here](https://regexper.com/#%28%5C%2B%7C-%29%3F%28inf%7CNaN%7C%28%5B0-9%5D%2B%7C%5B0-9%5D%2B%5C.%5B0-9%5D*%7C%5B0-9%5D*%5C.%5B0-9%5D%2B%29%28%28e%7CE%29%28%5C%2B%7C-%29%3F%5B0-9%5D%2B%29%3F%29).

Reply:
Also if people would prefer, I can migrate this to BNF, if people think that'd be easier for readers

Reply:
We could embed SVG of the railroad in-line… would look nice but barely maintainable.

Reply:
(Rust has [a library do draw these diagrams](https://crates.io/crates/railroad) too)